network_analysis/nbconverted/2_calculate_node_statistics.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Calculate node and edge statistics
+# 
+# In this script, we want to look in more detail at some of the graph attributes that might separate generic genes from non-generic genes. We'll look at:
+# 
+# * node (gene) degree in the eADAGE graph
+# * edge weight in the eADAGE graph
+# * [betweenness centrality](https://en.wikipedia.org/wiki/Betweenness_centrality) of generic vs. non-generic genes
+# * [PageRank](https://en.wikipedia.org/wiki/PageRank) (sometimes called PageRank centrality) of generic vs. non-generic genes, specifically the [undirected version](https://en.wikipedia.org/wiki/PageRank#PageRank_of_an_undirected_graph).
+
+# In[1]:
+
+
+import os
+
+import numpy as np
+import pandas as pd
+import graph_tool.all as gt
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+gt.seed_rng(1)
+np.random.seed(1)
+
+
+# In[2]:
+
+
+# relevant file paths
+data_dir = './data'
+processed_graph = os.path.join(data_dir, 'eadage_generic_graph_unsigned.gt')
+
+
+# In[3]:
+
+
+G = gt.load_graph(processed_graph)
+# make sure vertex/edge properties exist
+print(G)
+print(list(G.vp.keys()))
+print(list(G.ep.keys()))
+
+
+# In[4]:
+
+
+# plot degree distribution + look at degrees of generic genes
```

Comment:
SUPER nitpicky, but i would make this a markdown cell with this comment this this comment extends to multiple cells